Add info about Email/Password Auth (#382)
mongodb/docs-app-services: source/authentication/email-password.txt

Email/password authentication lets users register and login using an email address.
Atlas App Services must confirm Email/Password users before they may log in.

> Users may choose one of three ways to handle user confirmation:

Reviewer: Right above this, we're talking about "Users" in the context of end users of an app registering and logging in, but here I think you're talking about developers, so I would avoid "Users" here.

```suggestion
Developers may choose one of three ways to handle user confirmation:
```

Reviewer: After reading the page, I think having this "three ways to handle user confirmation" section in the Overview is a little strange. I would expect to find this down below in the "User Confirmation" section. (And I see that a summary _is_ there as I would expect.) I think of the Overview as more of a high-level description and the three ways of handling user confirmation seem like implementation details that have been sort of shoe-horned in here.

- :ref:`Send a Confirmation Email <auth-send-a-confirmation-email>`

- :ref:`Run a Confirmation Function <auth-run-a-confirmation-function>`

- :ref:`Automatically Confirm Users <auth-automatically-confirm-users>`

The following diagram shows the process flow for logging in:

> .. figure:: /images/email-password-login-flow.png

Reviewer: Love this diagram! But now I want relevant diagrams for _all_ the authentication providers. 🤔

User Confirmation
-----------------

> User confirmation occurs when a user wants to create a new Email/password 

Reviewer: I wonder if we want to reframe this in terms of what the developer must do here? i.e. something like:

> Before an email/password user can authenticate, your app must register and 
> confirm the user. Registering an email/password user creates a new user object. 
> App Services then requires this user to confirm the creation of this account. You
> can select one of three methods to confirm a new user account:

Author: Slightly reworded because AAS doesn't necessarily require the _user_ to confirm their account. 

> Before an email/password user can authenticate, your app must register and
confirm the user's new account. Registering an email/password user creates a 
new user object. App Services then requires confirmation of the account. You can 
select one of three methods to confirm a new user account:

account. Atlas App Services supports three methods for confirming a user account:

- Send a confirmation email
- Run a confirmation function

> - Automatically create the account with no validation.

Reviewer: For consistency, we should either use punctuation for all or none of these bullet points.

```suggestion
- Automatically create the account with no validation
```

.. warning:: 

>    Automatically creating accounts is useful when developing and testing an app, 
>    but you should not use it in a production app.

Reviewer: I wonder if we want to word this more strongly? Maybe something like: 
```suggestion
   Only use automatic confirmation when you are developing your app.
   Production applications should always use a secure confirmation process.
```

   * - ``pending``

>      - App Services changes the user's confirmation status to ``Pending Confirmation``.

Reviewer: I wonder if we want to add something about the `Pending` case relative to the email being registered, as in attempting to re-register the user while the confirmation is `Pending` will fail because the user's email is in the system, even though their account hasn't been confirmed. I've seen people get tripped up on that before.

Maybe we want to point them to the [Admin API endpoint where they can list pending users](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/docsp-27595/admin/api/v3/#tag/users/operation/adminListPendingUsers), to check for a pending user account in that case?

Technically, I suppose that applies to both "Send a user confirmation email" and "confirm with a custom function" so maybe there should be a separate section at the top level of the User Confirmation section describing the three possible states, implications for the email being "in use", and how a dev can use the Admin API to check for a pending user?

      }
   }

> Unless the custom function automatically confirms the user, you must provide a 

Reviewer: I think "you must provide a means for users to respond to the confirmation message that the function sends" is specific to this example, but I could see a developer interpreting it as "a confirmation function sends a message that the user must respond to" which isn't true inherently. It could be doing anything. Maybe we say instead something like:

> Unless the custom function automatically confirms the user, you must provide a
> means for the user to complete the confirmation process after the function fires.
> After completing any additional confirmation steps, call the SDK's Confirm User 
> method to finalize the user account creation on Atlas App Services.

Author: much better!
